front/src/features/tables/components/TagPopup.tsx

```diff
@@ -0,0 +1,39 @@
+import { ButtonIcon } from "@/components/atoms/ButtonIcon";
+import { Popup } from "@/components/atoms/Popup";
+import { TagLabel } from "@/components/atoms/TagLabel";
+import { faPlus } from "@fortawesome/free-solid-svg-icons";
+import { Tag } from "@prisma/client";
+import { useTagPopup } from "../hooks/useTagPopup";
+
+type TagPopupProps = {
+    logId: string;
+    addTagToDisplay: (tag: Tag) => void;
+};
+
+export const TagPopup = ({ logId, addTagToDisplay }: TagPopupProps) => {
+    const { isPopupOpen, openPopup, setIsPopupOpen, tags, selectTag } =
```

Comment:
this seems odd. Wouldn't `setIsPopUpOpen` be dependent on the `openPopup` call? I don't think we need this much state

Reply:
maybe it's needed but this seems a bit confusing to me

Reply:
The `openPopup` is dependent on `setIsOpenPopup`, but adds an event handler to prevent click propagation. I have replaced the function `setIsOpenPopup` with `closePopup` to stay consistent in the naming